src/OrbitBase/include/OrbitBase/ImmediateExecutor.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_IMMEDIATE_EXECUTOR_H_
+#define ORBIT_BASE_IMMEDIATE_EXECUTOR_H_
+
+#include <type_traits>
+
+#include "OrbitBase/Future.h"
+#include "OrbitBase/FutureHelpers.h"
+#include "OrbitBase/Promise.h"
+#include "OrbitBase/PromiseHelpers.h"
+
+namespace orbit_base {
+
+// ImmediateExecutor is a special type of executor that executes its action during the Schedule or
+// ScheduleAfter call. When called directly it's just a verbose abstraction around a standard
+// function call.
+//
+// Its usage only makes sense in combination with Future::Then.
+//
+// Example:
+// Future<std::string> result = thread_pool_->Schedule(/* whatever */);
+// ImmediateExecutor immediate_executor{};
+// Future<void> void_future = result.Then(&immediate_executor, [](const std::string& str) { (void)
+// str; }); main_thread_executor_->WaitFor(void_future); // WaitFor only works with Future<void>
```

Comment:
nit: formatting is off...

Reply:
Ah yes. Of course I blame clang-format. :innocent: 

Thanks for spotting this!